server/vertx-server/src/main/scala/sttp/tapir/server/vertx/streams/Pipe.scala

```diff
@@ -156,10 +157,12 @@ object Pipe {
         case Pong(payload) =>
           writeFrame(VertxWebSocketFrame.pongFrame(Buffer.buffer(payload)))
         case Close(statusCode, _) =>
-          socket.reject(statusCode)
+          socket.close(statusCode.toShort, new Handler[AsyncResult[Void]] {
+            def handle(x: AsyncResult[Void]): Unit = ()
+          })
```

Comment:
the purpose of `reject` is not to close a websocket, but as it's name suggest to reject, for example it the path is wrong https://github.com/eclipse-vertx/vert.x/blob/462e3ea5203dd710e52a84bf8fba5f5a5dc5eb46/src/main/java/io/vertx/core/http/ServerWebSocket.java#L121-L132

Reply:
The `Close` message is from `client` side or `server` side?

The SPEC says that the endpoint needs to send and receive a close message before closing the underlying connection: https://www.rfc-editor.org/rfc/rfc6455#section-5.5.1 So I guess a `reject` call is necessary before the socket is close

Reply:
Everything I read about reject is about refusing the websocket during the handshake: https://vertx.io/docs/4.0.3/apidocs/io/vertx/core/http/ServerWebSocket.html#reject-- you can also read the `setHandshake` function. 
I will track down what `statusCode` is used here and where the `Close` request comes from, maybe it's actually used during the handshake

Reply:
btw the `cats` implementation of websocket for vertx is throwing the same error I got: 

```
java.lang.IllegalStateException: Cannot reject WebSocket, it has already been written to
```

Reply:
It seems so as I quickly checked the code. Let's use `socket.close` then